### ISSUE TITLE
CompileStream doesn't accept single callback to end method

### ISSUE THREAD

**Issue by AdamHerrmann** [ivogabe/gulp-typescript#388]
I'm getting an error trying to use gulp-typescript with duplexify.  After some digging it seems that duplexify is calling CompileStream's end method with only a callback.  CompileStream.end() is pasing that callback as data to CompileStream.write(), which then throws an error because the callback isn't a Vinyl File.  Here is a short reproducer:

```javascript
const Vinyl    = require('vinyl');
const ts       = require('gulp-typescript');
const tsStream = ts();
const myFile   = new Vinyl({path: './test.ts', contents: new Buffer(`class MyClass {}`)});

tsStream.write(file);
tsStream.end(() => console.log('Finished'));
```

Error:

```
example/node_modules/gulp-typescript/release/main.js:35
        if (file.isNull()) {
                 ^

TypeError: file.isNull is not a function
    at CompileStream._write (example/node_modules/gulp-typescript/release/main.js:35:18)
    at CompileStream.end (example/node_modules/gulp-typescript/release/main.js:53:14)
    at Object.<anonymous> (example/gulp-test.js:23:10)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:456:32)
    at tryModuleLoad (module.js:415:12)
    at Function.Module._load (module.js:407:3)
    at Function.Module.runMain (module.js:575:10)
    at startup (node.js:160:18)
```

I checked node's built in streams and they appear to support passing only a callback to end:

```javascript
const fs       = require('fs');
const fsStream = fs.createWriteStream('temp');

fsStream.write('Some text');
fsStream.end(() => console.log('Finished'));
```

**Comment by ivogabe** [comment]
That sounds like a bug indeed, can you create a PR?

**Comment by AdamHerrmann** [comment]
Sure.

**Comment by ivogabe** [comment]
This has been fixed in #389; thanks @AdamHerrmann!
